app/auth/utils.py

```diff
@@ -76,6 +76,18 @@ def update_openrecords_user(form):
         Users,
         (current_user.guid, current_user.auth_user_type))
 
+    if current_user.default_agency != form.default_agency.data:
+        update_object(
```

Comment:
As far as I can tell, there is no simple way to get this done. For now two updates seems to work ok.